Upgrades: Domain Mapping: remove unsupported features from mapping page

dzver opened the pull request:
MX is incompatible with CNAME, and we can't add any DNS records, because there is no zone record for the mapping. As a result both features will not work with mappings without registration, or with subdomains of domains registered with us.

p2MSmN-4m3-p2

stephanethomas commented on client/my-sites/upgrades/domain-management/edit/mapped-domain.jsx:
```diff
@@ -69,15 +68,6 @@ const MappedDomain = React.createClass( {
-		return (
-			<VerticalNav>
-				{ this.emailNavItem() }
-				{ this.dnsRecordsNavItem() }
```
We should removed these two functions as well.

dzver commented:
Fixed

It's been awhile since my last PR here. It turned out that the vim eslint had stopped working. I fixed it. Now I see when I have unused vars and similar problems. Thanks.

stephanethomas commented:
Code looks good.
